argocd/structure_application_set.go

```diff
@@ -0,0 +1,1472 @@
+package argocd
+
+import (
+	"encoding/json"
```

Comment:
Nice. I wasn't aware of that project. Looks great. Will leave this as is for now since we use `encoding/json` in other places, so we can swap out the modules across the board in a separate PR.